lib/device-sockets/ios/socket-request-executor.ts

```diff
@@ -79,9 +74,18 @@ export class IOSSocketRequestExecutor implements IiOSSocketRequestExecutor {
 			await this.$iOSNotificationService.postNotification(deviceIdentifier, this.$iOSNotification.getAttachRequest(projectId, deviceIdentifier));
 			await readyForAttachPromise;
 		} catch (e) {
+			this.$logger.trace("Attach available error: ", e);
 			this.$errors.failWithoutHelp(`The application ${projectId} timed out when performing the socket handshake.`);
 		}
 	}
+
+	private async executeHandshake(deviceIdentifier: string, projectId: string, timeout: number): Promise<void> {
+		// This notification will be send only once by the runtime during application start.
+		// In case app is already running, we'll fail here as we'll not receive it.
+		const appLaunchingNotification = this.$iOSNotification.getAppLaunching(projectId);
+		const appLaunchingSocket = await this.$iOSNotificationService.postNotification(deviceIdentifier, appLaunchingNotification, constants.IOS_OBSERVE_NOTIFICATION_COMMAND_TYPE);
+		await this.$iOSNotificationService.awaitNotification(deviceIdentifier, +appLaunchingSocket, timeout);
```

Comment:
`+appLaunchingSocket` looks a bit strange, because `this.$iOSNotificationService.postNotification` should return a number according to the interface.

Reply:
I totally agree, but I just moved this code and prefer not changing it.